modal/cli/_download.py

```diff
@@ -37,7 +37,9 @@ async def producer():
             if is_pipe:
                 await q.put((None, entry))
             else:
-                output_path = local_destination / entry.path
+                start_path = os.path.dirname(remote_path).split("*")[0]
+                rel_path = os.path.relpath(entry.path, start_path.lstrip("/"))
+                output_path = local_destination / rel_path
```

Comment:
Can you use [Path from pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os.path`?

Reply:
Yeah entry.path and start_path are both strings though so I'd have to turn one into a PurePath